### PR TITLE
With sip phone support

### PR DESCRIPTION
**Pull request by rajinda**
Redesign existing web rtc and rest api phones to work with new common platform (strategy pattern).
Integrate agent console with sip phone.
Integrate agent dialer with sip phone.
Company configuration level SIP WEBRTC configuration
Initialize WEBRTC phone if communication with sip fails
show call log/contact list as new tab

https://duosoftware.atlassian.net/browse/DVV-1561

### CODE REVIEW

File: app/controller/console/consoleCtrl.js [hunk 1]
```diff
@@ -5,60 +5,60 @@
 
 agentApp.controller('consoleCtrl', function ($window, $filter, $rootScope, $scope, $http,
                                              $base64, $timeout, $q, $crypto, jwtHelper,
-                                             resourceService, baseUrls, dataParser, authService,
+                                             resourceService, baseUrls, shared_data, authService,
```
Comment (dinushadck) [inline app/controller/console/consoleCtrl.js]: duplicate usage of shared_data

File: app/services/common/shared_data.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Created by Rajinda Waruna on 25/04/2018.
+ */
+
+agentApp.factory('shared_data', function () {
+    return {agent_statue:"Offline", call_task_registered:false, currentModeOption:"",firstName:"",phone_config:{},callDetails: {},acw_time:10,phone_strategy:"",userProfile : {},userAccessFields : {}};//veery_web_rtc_phone
```
Comment (dinushadck) [inline app/services/common/shared_data.js]: typo should be 'agent_status' not 'agent_statue'

Comment (rajinda) [reply app/services/common/shared_data.js]: property name change to agent_status.

File: app/controller/console/consoleCtrl.js (hunk 1 above)
Comment (rajinda) [reply app/controller/console/consoleCtrl.js]: done

**Pull request merged**